src/renderer.tsx

```
@@ -287,11 +275,11 @@
   info: GPUCompilationInfo
 ) => {
   if (message == null) {
-    render("ok~", "Ok");
+    // renderTip("ok~", "Ok");
   } else {
     console.error(info);
     let before = code.split("\n").slice(0, info.messages[0].lineNum).join("\n");
     let space = " ".repeat(info.messages[0].linePos - 1);
-    render("error", before + "\n" + space + "^ " + message);
+    renderTip("error", before + "\n" + space + "^ " + message);
```

Biome

[error] Template literals are preferred over string concatenation.

Unsafe fix: Use a template literal.


(lint/style/useTemplate)